### PR TITLE
Change to new C++ based addon interface style and add fanart and screenshots

### PR DESCRIPTION
**Pull request by AlwinEsch**
This are changes related to Kodi request xbmc/xbmc#12262

Also thought to make the view on Kodi a bit nicer and added fanart and screenshots.
screenshot_20170608_053412

### CODE REVIEW

File: src/goom_config.h [hunk 1]
```diff
@@ -2,10 +2,10 @@
 #define GOOM_CONFIG
 
 #ifdef MID
-#define GOOM_TEXTURE_WIDTH  256
+#define GOOM_TEXTURE_WIDTH  455
```
Comment (akva2) [inline src/goom_config.h]: Peculiar sizes. 512x1024 is better for hw. Also it makes sense to out this as an option as this is software rendered and can eat some cpu (bot expecting you to do that bit)

Comment (AlwinEsch) [inline src/goom_config.h]: It can be changed, but must be something in 16/9
That's also the reason why I changed it, the ring before was not round :-D

Comment (akva2) [inline src/goom_config.h]: heh, how quaint. okay. at least make on of them a power-of-two so the other is easily handled with padding.

Comment (AlwinEsch) [inline src/goom_config.h]: What do you thing to use direct the resolutions 1280x720 and 640x360, sizes designed to match bit sizes.

With 512x455 you are right is really bad (455/8=56,875; 512/8=64)

With 1280x720 is it always divisible (720/8=90; 1280/8=160)
and 640x360 is always divisible (360/8=45; 640/8=80)

Comment (akva2) [inline src/goom_config.h]: my concern is igpu's and embedded devices which do not support non-power-of-two textures. they may not be very common these days, i don't really follow that part of the world, but they used to be. i know for sure that those MID devices did not. most of them will have efficient padding though, i.e. embedding the texture in the smallest possible overshoot.

File: src/goom_config.h [hunk 2]
```diff
@@ -2,11 +2,11 @@
...
-#define GOOM_TEXTURE_WIDTH  256
-#define GOOM_TEXTURE_HEIGHT 256
-#else  // MID
 #define GOOM_TEXTURE_WIDTH  512
-#define GOOM_TEXTURE_HEIGHT 512
+#define GOOM_TEXTURE_HEIGHT 256
+#else // MID
+#define GOOM_TEXTURE_WIDTH  1280
+#define GOOM_TEXTURE_HEIGHT 720
 #endif // MID
```
Comment (AlwinEsch) [inline src/goom_config.h]: Have replaced it, for `#ifdef MID` is it still the power of two.
The other use the standard screen resolution. Crazy was here during test that 512x512 and 1280x720 has nearly no difference in CPU usage. Still high ;-) as I've known it from GOOM for years.

Has looked in the source code, there is nothing described about best resolution and on his Test app they use 640x480 (http://goom.cvs.sourceforge.net/viewvc/goom/goom/test/testapp.c?view=markup)

Comment (akva2) [inline src/goom_config.h]: goom won't say anything about this. this is in the presentation layer - opengl stuff.

**Pull request merged**